File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        d1 = {}
        d2 = {}
        for x in s:
            if x in d1:
                d1[x] = d1[x] + 1
            else:
                d1[x] = 0
        
        for x in t:
            if x in d2:
                d2[x] = d2[x] + 1
            else:
                d2[x] = 0

        return d1 == d2

File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
#include <unordered_map>

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        int l = nums.size();

        std::unordered_map<int, int> dict;

        for(int i = 0; i < l; i++) {
            if (dict.count(nums[i])) {
                return true;
            }
            dict[nums[i]] = 0;
        }

        return false;
    }
};